### ISSUE TITLE
Spam Submissions produce Failed Queue Jobs

### ISSUE THREAD

**Issue by martinhellwagner** [solspace/craft3-freeform#201]
**Describe the bug or issue you're experiencing**

We're frequently getting Failed Queue Jobs for Spam Submissions in Freeform (see screenshot attached). Since we haven't found a way of disabling the building of search indexes for Spam Submissions, we're wondering how we can prevent these Queue Jobs from failing – is there anything we're missing?

**Steps to reproduce**

1. Receive Spam Submission in Freeform
2. Check Queue for Failed Job

**Expected behavior**

Spam Submissions shouldn't result in failed Queue Jobs

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.10
- Freeform Version: 3.11.3

**Additional context**

![Screenshot 2021-09-16 at 13 21 23](https://user-images.githubusercontent.com/19146752/133605843-7abc0423-c5b5-4c46-aa60-74cfaa00943b.png)

**Comment by kjmartens** [comment]
Sorry for the delay @martinhellwagner... does disabling the **Enable Search Index Updating on New Submissions?** setting make any difference for you? 🙂

**Comment by martinhellwagner** [comment]
It does, now there are no failed queue jobs any more. But can that really be the solution _not_ to index submission?

**Comment by kjmartens** [comment]
@martinhellwagner I am not suggesting this is the final solution, but at least it's a temporary workaround to prevent the errors from happening.

I haven't heard of anyone else reporting this issue yet, and I can't personally duplicate this in my testing. I wonder if it's something a bit more unique to your setup? Would you be able to let me know which other plugins you have installed (and if you know of any that specifically affect search indexes)? I'm wondering if there's anything being logged in the Craft / PHP error logs (`craft/storage/logs/phperrors.log`) as well... would you be able to check those? :)

**Comment by martinhellwagner** [comment]
@kjmartens 

As far as I'm aware, we have no plugin in our project that would interfere with Freeform submissions. Unfortunately, no errors pointing to the queue problem are logged either.

What's more, now I'm even getting the errors with the lightswitch "Enable Search Index Updating on New Submissions?" turned off. 🤔

**Comment by kjmartens** [comment]
I just have another question... how are you running the queue? Are you doing anything special like using a plugin, etc?

**Comment by martinhellwagner** [comment]
@kjmartens 

I'm a little confused why this has been closed. It has still not been resolved.

**Comment by kjmartens** [comment]
@martinhellwagner Sorry about that... closed it by accident while closing a couple of others. Can you try updating to 3.12 soon and see if it resolves the issue by chance? :)

**Comment by martinhellwagner** [comment]
@kjmartens 

Will check that!

**Comment by martinhellwagner** [comment]
@kjmartens 

One more question: since updating, our Sentry reports the following errors. Is it possible that we need to switch to "Encrypted Payload"?

![Screenshot 2021-11-19 at 09 55 14](https://user-images.githubusercontent.com/19146752/142594371-9b9135f3-ea4b-4f81-9ab7-9749a4af6ff0.png)

**Comment by martinhellwagner** [comment]
Update: I've switched to "Encrypted Payload", but the issue still persists. Any idea why this is happening?

**Comment by gustavs-gutmanis** [comment]
@martinhellwagner when you post a form, what does the `formHash` value look like?

**Comment by martinhellwagner** [comment]
@gustavs-gutmanis 

This is what the `formHash` looks like:

```
<input type="hidden" name="formHash" value="dAex9YxVN-XkmLnY8DA-7fwKFu9EettupGIecUgvfA5qVbtvxXMO1MOkAZL5" />
```

**Comment by gustavs-gutmanis** [comment]
That looks exactly like it should look.
Can you find out what it looks like before the error happens in Sentry?

**Comment by martinhellwagner** [comment]
I'm also seeing the problem in the Admin dashboard, so I'm not sure if it's really a problem with submitting a form:

![Screenshot 2021-11-19 at 11 28 15](https://user-images.githubusercontent.com/19146752/142607805-5e9ba0dc-a245-4215-b2e6-6c7c58827941.png)

**Comment by gustavs-gutmanis** [comment]
This is in a `try/catch` statement where if it doesn't get the correct hash, it instead returns the tuple of `null`s.
I'm guessing Sentry observes all unsuccessful `try` statements and reports them?
I can improve the check, but seems strange that it would report on cought exceptions

**Comment by martinhellwagner** [comment]
For what it's worth, the issue is only reported in our Sentry since we've updated to version 3.12 – is it possible that some change in this update started causing the problem?

**Comment by gustavs-gutmanis** [comment]
Yes this part was added in 3.12, so that's why it didn't happen before. I'll add a more resilient check there.

**Comment by martinhellwagner** [comment]
Thanks so much, I'll appreciate it! 🙏🏼

**Comment by gustavs-gutmanis** [comment]
It'll be released with the next FF version.

**Comment by kjmartens** [comment]
This is now fixed in Freeform 3.12.5. 🙂 

I'll close this issue for now. If the Failed Queue Jobs issue comes back, please just reply and we can re-open this issue. 🙂
